packages/js/components/src/experimental-select-control/select-control.tsx

```diff
@@ -152,12 +157,10 @@ function SelectControl< ItemType = DefaultItemType >( {
 		itemToString: getItemLabel,
 		onSelectedItemChange: ( { selectedItem } ) =>
 			selectedItem && onSelect( selectedItem ),
-		onInputValueChange: ( changes ) => {
-			if ( changes.inputValue !== undefined ) {
-				setInputValue( changes.inputValue );
-				if ( changes.isOpen ) {
-					onInputChange( changes.inputValue );
-				}
+		onInputValueChange: ( { inputValue: value, ...changes } ) => {
+			if ( value !== undefined ) {
+				setInputValue( value );
+				onInputChange( value, changes );
```

Comment:
@joshuatf note that I did end up updating this logic and exposing the changes as a second param.

I realized we did need this to be triggered all the time, especially in a async multi select, where a user uses there keyboard to select an item. The input value gets cleared afterwards, so if the user hits a down arrow after, the results of the original search value are still being shown, which is wrong.